arch/arm64/core/Kconfig

```diff
@@ -147,7 +147,10 @@ config ARM64_ENABLE_FRAME_POINTER
 	bool
 	default y
 	depends on OVERRIDE_FRAME_POINTER_DEFAULT && !OMIT_FRAME_POINTER
+	depends on !FRAME_POINTER
+	select DEPRECATED
```

Comment:
Won't this give a deprecation warning when building without `CONFIG_FRAME_POINTER` set? What if the user doesn't want this option enabled, is that possible?

Reply:
Yeah that is the intention.

This PR is trying to deprecate `CONFIG_<ARCH>_ENABLE_FRAME_POINTER` in favour of `CONFIG_FRAME_POINTER`. If the user wants frame pointer to be enabled, they should enable `CONFIG_FRAME_POINTER`.

Reply:
Yes, I know, but this `Kconfig` option is `default y`, so even if the user intentionally doesn't want to enable `CONFIG_FRAME_POINTER`, because it is user selectable, the warning will be present.

Reply:
This opens another question, updating removes the default feature now... 😛

Reply:
Hmm... Luckily we have the migration note for that 😜